### PR TITLE
What's New entry for threadsafety improvements

### PR DESCRIPTION
**Pull request by mhvk**
Trying to be ahead of the game (for once...).

**EDIT: Need these to get merged first:**

- [x] #11224 
- [x] #11226
- [x] #11227
- [x] #11234

### CODE REVIEW

File: docs/whatsnew/4.3.rst
Comment (pllim) [inline docs/whatsnew/4.3.rst:44]:
```suggestion
Improvements in making Astropy thread-safe
==========================================
```

File: docs/whatsnew/4.3.rst
Comment (pllim) [inline docs/whatsnew/4.3.rst:47]:
```suggestion
has been started to make parts of ``astropy`` thread-safe for operations that do not explicitly
```

File: docs/whatsnew/4.3.rst
Comment (pllim) [inline docs/whatsnew/4.3.rst:49]:
```suggestion
leap-second tables should now be thread-safe, as should any use of the
```

**Review by pllim — COMMENTED** [review]
Just minor comments. I will also add some checkboxes in your original post above, because we should not merge this until all the relevant PRs are merged, just in case.

**Comment by pllim** [comment]
Also cc @bmerry

File: docs/whatsnew/4.3.rst
Comment (bmerry) [inline docs/whatsnew/4.3.rst:46]:
```suggestion
To help use of Astropy in software for the Square Kilometre Array, an effort
```

Comment (bmerry) [reply docs/whatsnew/4.3.rst:46]: I wouldn't say it's just for the SKA though. I work on the MeerKAT radio telescope where we're also going to want these fixes. Maybe something like "To help use of Astropy in multi-threaded environments (for example, with Dask), ..."

File: docs/whatsnew/4.3.rst
Comment (bmerry) [inline docs/whatsnew/4.3.rst:48]:
```suggestion
change state. In particular, parsing of units and angles, and initializing
```
Since it's the automatic initialization that's a problem, rather than any explicit update made by the user (not sure if you prefer US or UK spelling).

Comment (pllim) [reply docs/whatsnew/4.3.rst:48]: > prefer US or UK spelling

⚔️ Came up on Slack, I don't think we agreed on anything about this... ⚔️

File: docs/whatsnew/4.3.rst
Comment (mhvk) [reply docs/whatsnew/4.3.rst:46]: Thanks, done

File: docs/whatsnew/4.3.rst
Comment (mhvk) [reply docs/whatsnew/4.3.rst:48]: Being in Canada I never know which one to use...

File: docs/whatsnew/4.3.rst
Comment (mhvk) [reply docs/whatsnew/4.3.rst:49]: Thanks, rephrased slightly.

File: docs/whatsnew/4.3.rst
Comment (mhvk) [reply docs/whatsnew/4.3.rst:47]: Mostly used - Used "has been started" but replaced "parts of"  with "the basic infrastructure of"

File: docs/whatsnew/4.3.rst
Comment (mhvk) [reply docs/whatsnew/4.3.rst:44]: done

**Comment by mhvk** [comment]
OK, rewritten following your comments - thanks, much clearer now!

**Review by pllim — APPROVED** [review]
All the PRs are merged and comments seem to be addressed, so I am approving, but I'll let you merge when you are ready, @mhvk . Thanks, all!

**Pull request merged**
